### PR TITLE
Videos table status

### PR DESCRIPTION
**Pull request by jtwray**
# Description
Adds new columns to the videos table migration
 - videoCompleted boolean 
 - thumbnail Url
Adjusts seeds to match new tables
 - marks the videos with mp4's as videoCreated
 - seeds in placeholder images
# Motivation
Thumbnails
 - offer a placeholder while the video is being built.
 - offers a more performant way to differentiate videos on the multivideos page. 
 - Provides a visual cue to tell them apart.
Adding the status column to our videos table.
 -  On true the front end will retrieve and display the video for user.

## Type of change
 
- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
    -- update the endpoints documentation.

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [x] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [x] Test A - ran migrations and seeded local database.
 data seeds as expected and thumbnails display / videos play

- [ ] Test B

# Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: api/api_router.js [hunk 1]
```diff
@@ -2,18 +2,16 @@ const router = require('express').Router();
 const restricted = require('../middleware/restricted_middleware');
 const authRouter = require('../auth/auth_router');
 const userRouter = require('../users/users_router');
-const video_update = require('../videos/video_update');
-const videos = require('../videos/videos');
-const songs = require('../songs/songs');
+const videosRouter = require('../videos/videos_router.js');
```
Comment (hisnameisjimmy) [inline api/api_router.js:5]: Yeah, makes it more clear

File: data/migrations/20200426223901_music.js [hunk 2]
```diff
@@ -17,6 +17,8 @@ exports.up = async function (knex, Promise) {
     video.increments();
     video.string("video_title").notNullable();
     video.string("location").notNullable();
+    video.boolean("videoCreated").defaultTo("false");
+    video.string("thumbnail");
```
Comment (hisnameisjimmy) [inline data/migrations/20200426223901_music.js:21]: I like using a boolean, but we probably want to stick with our naming convention, something like `video_created` instead

File: data/seeds/03_videos.js [hunk 3]
```diff
@@ -5,46 +5,133 @@ exports.seed = function (knex, Promise) {
       return knex("videos").insert([
         {
           video_title: "Testing",
-          location: "https://www.youtube.com/watch?v=82Q6DRqf9H4",
+          location:
+            "https://elasticbeanstalk-us-east-1-427400502172.s3.amazonaws.com/Videosforbackend/mayhem--push_every_button-testing--82Q6DRqf9H4.mp4",
```
Comment (hisnameisjimmy) [inline data/seeds/03_videos.js:9]: Here is the path to the actual DS files: 
https://artificial-artist.s3.amazonaws.com/

Here are the file names:
image

File: middleware/checkfor.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+checkFor([a,b,c,d,e])
```
Comment (hisnameisjimmy) [inline middleware/checkfor.js:1]: This is interesting, this is what @panamajake86 was potentially looking at doing. Might want to check in with him.

File: videos/videos_model.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
+const db = require("../data/db_config");
```
Comment (hisnameisjimmy) [inline videos/videos_model.js:1]: Good in keeping the naming conventions solid here, but now I'm gonna hit a merge conflict with your work :)

File: videos/videos_model.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  const [id] = await db("videos").insert(data, "id");
+
+  return id;
+  // return findById(id);
+}
+
+function find9() {
+  const res= db.raw("select * from videos ORDER BY random() limit 9 ");
+  return res;
+}
```
Comment (hisnameisjimmy) [inline videos/videos_model.js:23]: This is pretty rad, nice SQL query

File: videos/videos_router.js [hunk 7]
```diff
@@ -13,6 +13,15 @@ router.get("/", async (req, res) => {
   }
 });
 
+router.get("/random9", async (req, res) => {
+  try {
+    const {rows} = await Videos.find9();
+    res.status(200).json({rows});
+  } catch ({message}) {
+    res.status(500).json({errorMessage: `Encountered |*||${message}|*| while retrieving videos from the database.`});
```
Comment (hisnameisjimmy) [inline videos/videos_router.js:21]: What would this message be? Should this be `err`?

File: data/migrations/20200426223901_music.js (hunk 2 above)
Comment (jtwray) [reply data/migrations/20200426223901_music.js:21]: Good catch!

File: data/seeds/03_videos.js (hunk 3 above)
Comment (jtwray) [reply data/seeds/03_videos.js:9]: Excellent, Will get these in.

File: videos/videos_router.js (hunk 7 above)
Comment (jtwray) [reply videos/videos_router.js:21]: Err would be more declarative, wouldn't it.
 Sounds good.

File: videos/videos_model.js (hunk 6 above)
Comment (jtwray) [reply videos/videos_model.js:23]: 🚀

File: videos/videos_router.js (hunk 7 above)
Comment (jtwray) [reply videos/videos_router.js]: updates response variable at this commit  https://github.com/Lambda-School-Labs/artificial-artist-be/pull/23/commits/e3b43a185cc003abbb56df91647f044b687e5062

File: data/migrations/20200426223901_music.js (hunk 2 above)
Comment (jtwray) [reply data/migrations/20200426223901_music.js]: updates the migration to match naming convention at this commit https://github.com/Lambda-School-Labs/artificial-artist-be/pull/23/commits/efdfcbdc5ee5831119b99b3d0bf09e26cd7a0a39

File: data/seeds/03_videos.js (hunk 3 above)
Comment (jtwray) [reply data/seeds/03_videos.js]: revisits the seed file to include the correct S3 bucket and Actual created videos here https://github.com/Lambda-School-Labs/artificial-artist-be/pull/23/commits/ed692ff7bdc86e087bf5959642bb319b8a814096

File: middleware/checkfor.js (hunk 4 above)
Comment (jtwray) [reply middleware/checkfor.js:1]: I met with @panamajake86 today we came up with a clean middle ware function for this one.  will followup pr

**Pull request merged**